app/services/foreman_bootdisk/renderer_methods.rb

```diff
@@ -10,14 +10,18 @@ def format_bootdisk_chain_url(url, mac)
         u.fragment = nil
         u.to_s
       end
+
+      def bootdisk_raise(*args)
+        raise ::Foreman::Exception.new(*args)
+      end
     end
 
-    def bootdisk_chain_url(mac = '', action = 'iPXE')
+    def bootdisk_chain_url(mac = self.try(:mac), action = 'iPXE')
       self.class.format_bootdisk_chain_url(foreman_url(action), mac)
     end
 
     def bootdisk_raise(*args)
-      raise ::Foreman::Exception.new(*args)
+      self.class.bootdisk_raise(*args)
```

Comment:
Is this really required? Where is `Host.bootdisk_raise` called?

Reply:
It's not really, I did it for consistency. Removing.

Reply:
Oh wait it is needed, it's controller class vs instance. We need both. When testing, try it without this.